pyproject.toml

```diff
@@ -55,7 +55,7 @@ maintainers = [
 # Note: Python and NumPy upper version bounds should be set correctly in
 # release branches, see:
 #     https://scipy.github.io/devdocs/dev/core-dev/index.html#version-ranges-for-numpy-and-other-dependencies
-requires-python = ">=3.8"
+requires-python = ">=3.8,<3.11"
```

Comment:
do we really want to set an upper bound in `master` branch?

Reply:
actually, this is already set on the release branch: https://github.com/scipy/scipy/blob/maintenance/1.8.x/pyproject.toml#L58

So we need to back this change out, but the rest seems "ok" to me.

Reply:
I was just following what was done on the 1.7 branch: https://github.com/scipy/scipy/pull/14921.  If we don't want to upper bound master, should I make a PR to the 1.8 branch to pin it?

Reply:
No, because it is already done on the `maintenance/1.8.x` branch as I linked above.